packages/astro/src/vite-plugin-build-html/index.ts

```diff
@@ -452,5 +456,30 @@ export function rollupPluginAstroBuildHTML(options: PluginOptions): VitePlugin {
         });
       }
     },
-  };
+  }, {
+    name: '@astro/rollup-plugin-build-load-fallback',
+    enforce: 'post',
+    resolveId(id) {
+      // If we got here, nothing picked up this module, so let's try
+      // and grab it from the viteServer used for SSR loading.
+      if(id[0] === '/') {
+        return id;
+      }
+      return null;
+    },
+    async load(id) {
+      const init: RequestInit = {
+        headers: {
+          accept: mime.getType(id)
+        }
+      };
+
+      const res = await fetch(id, init);
```

Comment:
This is where the fix is. Since there is no client-side usage of uno.css, only SSR, the CSS for this module is in the `viteServer`. So we need to fetch() against it to get the result.